pkg/bootstrap/cni.go

```diff
@@ -82,6 +83,11 @@ func internalRenderCNI(ctx context.Context, cl client.Client, config cni.Config,
 	values := make(map[string]interface{})
 	values["cni"] = cni
 
+	serverVersion, err := dc.ServerVersion()
```

Comment:
Unfortunately, the server version can change while the operator is alive. I'm not sure if the discoveryClient that is passed here is the cachedDiscoveryClient or not. 

However, even if we're hitting the API server every time, it's not that big of a deal, because we perform far more requests that update the resources owned by the SMCP.